src/Debugging/Debugger.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Debugging;
+
+use InvalidArgumentException;
+use Saloon\Contracts\DebuggingDriver;
+use Saloon\Debugging\Drivers\RayDebugger;
+use Saloon\Debugging\Drivers\ErrorLogDebugger;
+use Saloon\Debugging\Drivers\SystemLogDebugger;
+use Saloon\Exceptions\UnknownDriverException;
+
+class Debugger
+{
+    /**
+     * Globally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected static array $globalRegisteredDrivers = [];
+
+    /**
+     * Locally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected array $registeredDrivers = [];
+
+    /**
+     * Drivers that have been subscribed to
+     *
+     * @var array<string, bool>
+     */
+    protected array $useDrivers = [];
+
+    /**
+     * Denotes if we send the request to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showRequest = false;
+
+    /**
+     * Denotes if we send the response to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showResponse = false;
+
+    public function __construct()
+    {
+        $this->registerDriver(new RayDebugger);
+        $this->registerDriver(new ErrorLogDebugger);
+        $this->registerDriver(new SystemLogDebugger);
+    }
+
+    /**
+     * Register a driver globally
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return $this
```

Comment:
Just saw this in the PHPStan errors. 🙂 

```suggestion
     * @return static
```